bindings/go/src/fdb/fdb.go

```diff
@@ -195,48 +194,74 @@ func MustGetAPIVersion() int {
 var apiVersion int
 var networkStarted bool
 var networkMutex sync.RWMutex
+var networkRunning sync.WaitGroup
 
 var openDatabases sync.Map
 
-func startNetwork() error {
+// ensureNetworkIsStarted ensures that the API version is set and then starts the internal network event loop, if not already done.
+// The provided function is guaranteed to be run while network thread is running.
+func ensureNetworkIsStarted(f func()) error {
 	networkMutex.RLock()
 	if networkStarted {
+
+		// network thread is guaranteed to be running while this user-provided function runs
+		f()
+
 		networkMutex.RUnlock()
 		return nil
 	}
 	// release read lock and acquire write lock
 	networkMutex.RUnlock()
 	networkMutex.Lock()
 	defer networkMutex.Unlock()
-	if networkStarted {
-		return nil
-	}
 
-	if e := C.fdb_setup_network(); e != 0 {
-		return Error{int(e)}
-	}
-
-	go func() {
-		e := C.fdb_run_network()
-		if e != 0 {
-			log.Printf("Unhandled error in FoundationDB network thread: %v (%v)\n", C.GoString(C.fdb_get_error(e)), e)
+	// check if meanwhile another goroutine started the network thread
+	if !networkStarted {
```

Comment:
Instead of having the RWMutex implementation, do you think using [Once](https://pkg.go.dev/sync#Once) would be a better (easier to read) version of this?

Reply:
I think it's not possible because the mutex is currently used to protect various operations which interact with the network thread. The "R" in this RWMutex is there just to allow a fast exit in case network is already started.